tests/integration/models/device.spec.coffee

```diff
@@ -753,6 +753,208 @@ describe 'Device Model', ->
+						resource: 'image'
+						body:
+							is_part_of__release: oldRelease.id
+							is_a_build_of__service: webService.id
+							project_type: 'dockerfile'
+							content_hash: 'abc'
+							start_timestamp: 1234
+							status: 'success'
+				,
+					resin.pine.post
+						resource: 'image'
+						body:
+							is_part_of__release: newRelease.id
+							is_a_build_of__service: webService.id
+							project_type: 'dockerfile'
+							content_hash: 'def'
+							start_timestamp: 54321
+							status: 'success'
+				,
+					resin.pine.post
+						resource: 'image'
+						body:
+							is_part_of__release: newRelease.id
+							is_a_build_of__service: dbService.id
+							project_type: 'dockerfile'
+							content_hash: 'ghi'
+							start_timestamp: 54321
+							status: 'success'
+				]
+				.spread (device, oldWebImage, newWebImage, dbImage) ->
+					Promise.all [
+						# Create image installs for the images on the device
+						resin.pine.post
+							resource: 'image_install'
+							body:
+								installs__image: oldWebImage.id
+								is_provided_by__release: oldRelease.id
+								device: device.id
+								download_progress: 100
+								status: 'running'
+								install_date: '2017-10-01'
+					,
+						resin.pine.post
+							resource: 'image_install'
+							body:
+								installs__image: newWebImage.id
+								is_provided_by__release: newRelease.id
+								device: device.id
+								download_progress: 50,
+								status: 'downloading'
+								install_date: '2017-10-30'
+					,
+						resin.pine.post
+							resource: 'image_install'
+							body:
+								installs__image: dbImage.id
+								is_provided_by__release: newRelease.id
+								device: device.id
+								download_progress: 100,
+								status: 'running',
+								install_date: '2017-10-30'
+					,
+						# Create service installs for the services running on the device
+						resin.pine.post
+							resource: 'service_install'
+							body:
+								installs__service: webService.id
+								device: device.id
+					,
+						resin.pine.post
+							resource: 'service_install'
+							body:
+								installs__service: dbService.id
+								device: device.id
+					]
+
+		describe 'resin.models.device.getWithServiceDetails()', ->
+
+			it 'should be able to get the device by uuid', ->
+				resin.models.device.getWithServiceDetails(@device.uuid).then (device) =>
+					m.chai.expect(device.id).to.equal(@device.id)
+
+			it 'should be able to get the device by id', ->
+				resin.models.device.getWithServiceDetails(@device.id).then (device) =>
+					m.chai.expect(device.id).to.equal(@device.id)
+
+			it 'should retrieve the current service details', ->
+				resin.models.device.getWithServiceDetails(@device.id).then (deviceDetails) =>
+					m.chai.expect(deviceDetails).to.deep.match
+						device_name: @device.device_name
+						uuid: @device.uuid
+						current_release: @currentRelease.commit
+						current_services:
+							web: [
+								commit: 'new-release-commit'
```

Comment:
It would be really handy to have the service install id retrieved here, so that we can key the container elements correctly in the UI.

Reply:
Do you mean the service install or the image install? Service installs should, I think, be unique by device + name, so this should be sufficient already. Is there an annoying special case somewhere where that's not true?

Reply:
ah you're right, I need the image install id, because you can have multiple image installs on a device with the same device+name combination.

Reply:
Bonus: this is actually already there conveniently (as `id`, in each of these objects), the tests just didn't check for it. The tests now do check for it, since you're right that's totally useful, and you should be able to start using it immediately regardless.